qiskit/algorithms/optimizers/qnspsa.py

```diff
@@ -95,6 +92,7 @@ def __init__(
         lse_solver: Optional[Callable[[np.ndarray, np.ndarray], np.ndarray]] = None,
         initial_hessian: Optional[np.ndarray] = None,
         callback: Optional[CALLBACK] = None,
+        termination_checker: Optional[CALLBACK] = None,
```

Comment:
Isn't there a difference between these - the `callback` has no return ie is

`Callable[[np.ndarray, float, float, int, bool], None]`

whereas the `termination_checker` returns a bool ie

`Callable[[np.ndarray, float, float, int, bool], bool]`

So while the signatures are very similar, they have this return type difference

Reply:
@woodsp-ibm You are right. I updated the annotation.